Rewrite extension management tools

angelahning opened the pull request:
## Summary
<!-- Describe your change -->
This PR refactors extension management tools from being hardcoded in the Agent into a proper platform extension (ExtensionManagerClient), allowing it to be enabled/disabled like any other extension.

### Type of Change
<!-- Select all that apply -->
- [x] Feature
- [ ] Bug fix
- [x] Refactor / Code quality
- [ ] Performance improvement
- [ ] Documentation
- [ ] Tests
- [ ] Security fix
- [ ] Build / Release
- [ ] Other (specify below)

### Testing
<!-- How have this change been tested? Unit/integration tests? Manual testing? -->
Manually tested on UI and CLI

### Related Issues
Addesses https://github.com/block/goose/issues/2331


### Screenshots/Demos (for UX changes)
Before:  

After: 
Agent should not have access to the extension management tools when the extension_manager extension is disabled 
<img width="921" height="653" alt="Screenshot 2025-10-07 at 2 49 08 PM" src="https://github.com/user-attachments/assets/6b677770-f9fe-4181-94f9-8596db545dce" />

Re-enabled the extension, agent can now autonomously discover and enable/disable tools.
<img width="719" height="628" alt="Screenshot 2025-10-07 at 2 49 49 PM" src="https://github.com/user-attachments/assets/7e8b6d97-a3ae-4b28-ac2b-6f2cf6db5ef6" />
<img width="722" height="635" alt="Screenshot 2025-10-07 at 2 49 59 PM" src="https://github.com/user-attachments/assets/91f75528-6a97-4071-acae-429da35152aa" />

angelahning commented on crates/goose/src/agents/extension_manager_extension.rs:
```diff
@@ -0,0 +1,535 @@
+use tracing::error;
+
+pub static EXTENSION_NAME: &str = "extensionmanager";
+pub static DISPLAY_NAME: &str = "Extension Manager";
```
somewhere in the code keeps modify the name "Extension Manager" to "extensionmanager", and then failing with a Unknown platform extension error. I spent sometime figuring out why that is, and decided it is faster to just add a display name like the built-ins.

DOsinga replied:
yeah, I don't think that's a good reason. having names and display names is in general a recipe for confusion and mismatches, if we can't track them down ourselves that's even worse.

angelahning replied:
i will track it down

angelahning commented:
- updated system prompt
- added unit tests
- the router stuff should be its own tool, not migrated in this PR
- fixed existing bug in UI where it was mutating the extenstion name
